cmd/swarmctl/service/flagparser/secret.go

```diff
@@ -0,0 +1,189 @@
+package flagparser
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/cmd/swarmctl/common"
+	"github.com/spf13/cobra"
+)
+
+func parseSecretString(secretString string) (secretName, versionID, presentName string, err error) {
```

Comment:
Comment was added above the function:  `// expects secrets in the format SECRET_NAME:TARGET_NAME`